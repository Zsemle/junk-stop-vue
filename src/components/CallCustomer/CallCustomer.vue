<template>
  <div class="call-customer">
    <template v-if="phoneNumber">
      <a title="Call the customer" :href=phoneLink>Call customer</a>
    </template>
    <span v-else>phone number unavailable</span>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'StopOrder',
  props: {
    phoneNumber: {
      default: null,
      type: String,
    },
  },
  computed: {
    phoneLink() {
      return `tel:${this.phoneNumber}`;
    },
  },

});
</script>

<style lang="scss" scoped>
.stop-order{
  border: 1px solid rgb(51, 128, 0);
}
</style>
