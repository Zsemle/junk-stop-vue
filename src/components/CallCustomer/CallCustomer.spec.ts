import { shallowMount } from '@vue/test-utils';
import CallCustomer from './CallCustomer.vue';

describe('AppFrame.vue', () => {
  it('renders call link if phone number is available', () => {
    const props = {
      phoneNumber: '0612345678',
    };
    const wrapper = shallowMount(CallCustomer, {
      propsData: props,
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders plain text if phone number is not available', () => {
    const wrapper = shallowMount(CallCustomer);
    expect(wrapper).toMatchSnapshot();
  });
});
