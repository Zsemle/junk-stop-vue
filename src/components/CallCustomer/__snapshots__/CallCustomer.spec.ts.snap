// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFrame.vue renders call link if phone number is available 1`] = `<div class="call-customer"><a title="Call the customer" href="tel:0612345678">Call customer</a></div>`;

exports[`AppFrame.vue renders plain text if phone number is not available 1`] = `<div class="call-customer"><span>phone number unavailable</span></div>`;
